core/DataTable/Manager.php

```diff
@@ -12,30 +12,31 @@
 use Exception;
 use Piwik\Common;
 use Piwik\DataTable;
-use Piwik\Singleton;
 
 /**
  * The DataTable_Manager registers all the instanciated DataTable and provides an
  * easy way to access them. This is used to store all the DataTable during the archiving process.
  * At the end of archiving, the ArchiveProcessor will read the stored datatable and record them in the DB.
- *
- * @method static \Piwik\DataTable\Manager getInstance()
  */
-class Manager extends Singleton
+class Manager extends \ArrayObject
```

Comment:
An implementation based on `ArrayObject` is much faster. Eg when having many `getTable` calls this can save 60 seconds or 3% wall time